packages/at_onboarding_flutter/lib/screen/at_onboarding_activate_screen.dart

```diff
@@ -192,12 +192,22 @@ class _AtOnboardingActivateScreenState
         return;
       }
 
-      //Delay 10s for waiting for ServerStatus change to teapot when activating an atsign
+      //Delay for waiting for ServerStatus change to teapot when activating an atsign
       await Future.delayed(const Duration(seconds: 10));
+
       _onboardingService.setAtClientPreference =
           widget.config.atClientPreference;
+
       authResponse = await _onboardingService.authenticate(atsign,
           cramSecret: secret, status: OnboardingStatus.ACTIVATE);
+
+      atSignStatus = await _onboardingService.checkAtSignServerStatus(atsign);
+      while (atSignStatus != ServerStatus.activated) {
+        await Future.delayed(const Duration(seconds: 3));
+        atSignStatus = await _onboardingService.checkAtSignServerStatus(atsign);
+        debugPrint("currentAtSignStatus: $atSignStatus");
+      }
```

Comment:
Can you please add a max wait time to 30 seconds?

Reply:
I've fixed it.